lib/scan_beacon/interleaved_advertiser.rb

```diff
@@ -0,0 +1,61 @@
+module ScanBeacon
+  class InterleavedAdvertiser < GenericAdvertiser
+    attr_accessor :beacons, :parsers, :advertiser
+    # We will cycle through each beacon sequentially over the course of the time below
+    INTERLEAVE_CYCLE_MILLIS = 1000
+    
+    def initialize(opts = {})
+      super(opts)
+      self.advertiser = opts[:advertiser]
+      raise "No available advertiser" if advertiser.nil?
+      self.beacons = opts[:beacons]
+      if opts[:parsers]
+        self.parsers = opts[:parsers]
+      else
+        self.parsers = []
+        beacons.each do |beacon|
+          self.parsers << BeaconParser.default_parsers.find {|parser| parser.beacon_type == beacon.beacon_types.first}
+        end
+      end
+      raise "You must supply the same number of beacons (#{beacons.count}) and parsers (#{parsers.count})" if beacons.count != parsers.count
+    end
+
+    def start
+      if beacons.size > 0
+        start_interleaving_thread
+      end
+    end
+
+    def stop
+      if @thread
+        @stop_requested = true
+        @thread.join
+        @thread = nil      
+        advertiser.stop
+      end
+    end
+
+    def inspect
+      "<InterleavedAdvertiser beacons=#{@beacons.inspect}>"
+    end
+
+  private
+    def start_interleaving_thread
+      sleep_time = 1000.0/INTERLEAVE_CYCLE_MILLIS/beacons.size
+      @stop_requested = false
+      @thread = Thread.new do
+        while !@stop_requested do
+          beacons.each_with_index do |beacon, index|
+            advertiser.stop
+            break if @stop_requested
+            advertiser.beacon = beacon
+            advertiser.parser = parsers[index]
+            advertiser.start
+            sleep sleep_time        
+          end
+        end
+      end
+    end    
```

Comment:
I see that the `BLE112Device` class doesn't provide a way to just set the advertisement data - it is combined with the method to start advertising.  We should probably break that out into a separate method so that we can simply swap in new advertisements while advertising.  I believe the BlueZ code already has this broken out.